bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java

```diff
@@ -0,0 +1,656 @@
+            try {
+                value = getValue(channelId, rthermData);
+            } catch (Exception e) {
+                logger.debug("Error setting {} value", channelId.toUpperCase());
+                return;
+            }
+
+            State state = null;
+            if (value == null) {
+                state = UnDefType.UNDEF;
+            } else if (value instanceof PointType) {
+                state = (PointType) value;
+            } else if (value instanceof ZonedDateTime) {
+                state = new DateTimeType((ZonedDateTime) value);
+            } else if (value instanceof QuantityType<?>) {
+                state = (QuantityType<?>) value;
+            } else if (value instanceof BigDecimal) {
+                state = new DecimalType((BigDecimal) value);
+            } else if (value instanceof Integer) {
+                state = new DecimalType(BigDecimal.valueOf(((Integer) value).longValue()));
+            } else if (value instanceof String) {
+                state = new StringType(value.toString());
+            } else if (value instanceof OnOffType) {
+                state = (OnOffType) value;
+            } else {
+                logger.warn("Update channel {}: Unsupported value type {}", channelId,
+                        value.getClass().getSimpleName());
+            }
+            logger.debug("Update channel {} with state {} ({})", channelId, (state == null) ? "null" : state.toString(),
+                    (value == null) ? "null" : value.getClass().getSimpleName());
+
+            // Update the channel
+            if (state != null) {
+                updateState(channelId, state);
+            }
+        }
+    }
+
+    /**
+     * Build request URL from configuration data
+     *
+     * @return a valid URL for the thermostat's JSON interface
+     */
+    private String buildRequestURL(String resource) {
+        RadioThermostatConfiguration config = getConfigAs(RadioThermostatConfiguration.class);
+
+        String hostName = StringUtils.trimToEmpty(config.hostName);
+
+        String urlStr = URL.replace("%hostName%", hostName);
+        urlStr = urlStr.replace("%resource%", resource);
+
+        return urlStr;
+    }
+
+    /**
+     * Request new data from the thermostat
+     *
+     * @param the resource URL constant for a particular thermostat JSON resource
+     * @return an object mapping to one of the various thermostat JSON responses or null in case of error
+     */
+    private Object getRadioThermostatData(String resource) {
+        Object result = null;
+        String errorMsg = null;
+
+        String urlStr = buildRequestURL(resource);
+        logger.debug("URL = {}", urlStr);
+
+        try {
+            // Run the HTTP request and get the JSON response from the thermostat
+            URL url = new URL(urlStr);
+            URLConnection connection = url.openConnection();
+
+            try {
+                String response = IOUtils.toString(connection.getInputStream());
```

Comment:
Jetty is the preferred library to perform HTTP requests: https://www.openhab.org/docs/developer/guidelines.html#default-libraries